src/eredis_client.erl

```diff
@@ -606,12 +606,12 @@ read_database(Database) when is_integer(Database) ->
 
 -spec get_auth_command(Username :: iodata() | undefined,
                        Password :: iodata() | undefined) ->
-          iodata() | undefined.
+          fun(() -> iodata()) | undefined.
 get_auth_command(undefined, undefined) ->
     undefined;
 get_auth_command(undefined, "") -> % legacy
     undefined;
 get_auth_command(undefined, Password) ->
-    eredis:create_multibulk([<<"AUTH">>, Password]);
+    fun () -> eredis:create_multibulk([<<"AUTH">>, Password]) end;
 get_auth_command(Username, Password) ->
-    eredis:create_multibulk([<<"AUTH">>, Username, Password]).
+    fun () -> eredis:create_multibulk([<<"AUTH">>, Username, Password]) end.
```

Comment:
This is slightly inefficient, because `eredis:create_multibulk/1` will be invoked every time the AuthCmd needs to be used, when we only need to do it once.

Maybe like this?

```erl
    get_auth_command(Username, Password) ->
        AuthCmd = eredis:create_multibulk([<<"AUTH">>, Username, Password]),
        fun () -> AuthCmd end.
```